Source/JavaScriptCore/jit/ExecutableAllocator.cpp

```diff
@@ -446,19 +458,21 @@ class FixedVMPoolExecutableAllocator final {
         m_reservation = WTFMove(reservation.pageReservation);
         if (m_reservation) {
 #if ENABLE(JUMP_ISLANDS)
+            constexpr size_t maxRegionSize = MacroAssembler::nearJumpRange;
+
             // These sizes guarantee that any jump within an island can jump forwards or backwards
             // to the adjacent island in a single instruction.
-            const size_t islandRegionSize = roundUpToMultipleOf(pageSize(), static_cast<size_t>(MacroAssembler::nearJumpRange * islandRegionSizeFraction));
-            m_regionSize = MacroAssembler::nearJumpRange - islandRegionSize;
-            RELEASE_ASSERT(isPageAligned(islandRegionSize));
-            RELEASE_ASSERT(isPageAligned(m_regionSize));
-            const unsigned numAllocators = (reservation.size + m_regionSize - 1) / m_regionSize;
+            const size_t islandRegionSize = roundUpToMultipleOf(EXECUTABLE_PAGE_SIZE, static_cast<size_t>(maxRegionSize * islandRegionSizeFraction));
+            m_regionSize = maxRegionSize - islandRegionSize;
```

Comment:
I believe this change is wrong for the reason we talked about in slack. But commenting here for posterity.

What this change does is make the layout:
```
[  island 0 ] [ JIT region          ] [ island 1 ]
-16          0                     112           128
```

But it should be:
```
[  island 0 ] [ JIT region          ] [ island 1 ]
0           16                      112          128
```

So any jump in island 0 can make it to island 1 using one instruction.

Reply:
Undid the erroneous "fix" and added a comment to document and clarify this detail instead.